### PR TITLE
Support for detailed error message in puppet-polymer-client

### PR DESCRIPTION
**Pull request by warpech**
Needed for https://github.com/PuppetJs/puppet-polymer-client/pull/32

@joozek78 pls review.

### CODE REVIEW

**Comment by joozek78** [comment]
Apart from [what I've already said in a different PR](https://github.com/PuppetJs/puppet-polymer-client/pull/32#discussion_r77131246), it's fine for me

**Comment by warpech** [comment]
I have made a change that makes the "data" schema more consistent.

File: src/puppet.js [hunk 1]
```diff
@@ -439,8 +439,7 @@
       var res = this;
       that.handleResponseHeader(res);
       if (res.status >= 400 && res.status <= 599) {
-        that.onFatalError(JSON.stringify({ statusCode: res.status, statusText: res.statusText, text: res.responseText }), url, method);
-        throw new Error('PuppetJs JSON response error. Server responded with error ' + res.status + ' ' + res.statusText + '\n\n' + res.responseText);
+        that.onFatalError({ statusCode: res.status, statusText: res.statusText, reason: res.responseText }, url, method);
```
Comment (joozek78) [inline src/puppet.js]: can you include those fields somewhere in the docs? Probably in ctor docstring (by the `onFatalError` callback description)

Comment (warpech) [inline src/puppet.js]: Done, thx for noticing

**Pull request merged**